### PR TITLE
Trigger reconciler tests for ConsumerGroup scaling

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: aavarghese <avarghese@us.ibm.com>

Part of #1537

/cc @pierDipi 

## Proposed Changes

- Adding new Trigger reconciler unit tests for consumer scheduling and scaling

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
I did a quick review of this PR, looks good overall.

File: control-plane/pkg/reconciler/trigger/v2/triggerv2.go [hunk 1]
```diff
@@ -83,25 +83,33 @@ func (r *ReconcilerV2) ReconcileKind(ctx context.Context, trigger *eventing.Trig
...
-
-	trigger.Status.SubscriberURI = consgroup.Status.SubscriberURI
+	if cg.IsReady() {
+		trigger.Status.MarkDependencySucceeded()
+	} else {
+		topLevelCondition := cg.GetConditionSet().Manage(cg.GetStatus()).GetTopLevelCondition()
+		if topLevelCondition == nil {
+			trigger.Status.MarkDependencyUnknown("failed to reconcile consumer group", "consumer group not ready")
+		} else {
+			trigger.Status.MarkDependencyFailed(topLevelCondition.Reason, topLevelCondition.Message)
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/trigger/v2/triggerv2.go:98]: Can we add another test to test this condition propagation?

File: control-plane/pkg/reconciler/trigger/v2/triggerv2.go [hunk 2]
```diff
@@ -83,25 +83,33 @@ func (r *ReconcilerV2) ReconcileKind(ctx context.Context, trigger *eventing.Trig
...
 	trigger.Status.MarkDeadLetterSinkResolvedSucceeded()
 
 	return nil
 }
 
-func (r ReconcilerV2) reconcileConsumerGroup(ctx context.Context, trigger *eventing.Trigger) (*internalscg.ConsumerGroup, error) {
+func (r Reconciler) reconcileConsumerGroup(ctx context.Context, trigger *eventing.Trigger) (*internalscg.ConsumerGroup, error) {
 
-	var deliveryOrdering = internals.Unordered
+	var deliveryOrdering = internals.Ordered
```
[the remark this answers is not in the document]
Comment (aavarghese) [reply control-plane/pkg/reconciler/trigger/v2/triggerv2.go:112]: Fixed. Thanks.

File: control-plane/pkg/reconciler/trigger/v2/triggerv2.go (hunk 1 above)
Comment (aavarghese) [reply control-plane/pkg/reconciler/trigger/v2/triggerv2.go:98]: Done

**Review by pierDipi — APPROVED** [review]
Thank you! LGTM

/lgtm
/approve

**Pull request merged**